scalameta/scalameta/src/test/scala/scala/meta/tests/parsers/XmlSuite.scala

```diff
@@ -0,0 +1,148 @@
+package scala.meta.tests.parsers
+
+import scala.meta._
+import scala.meta.dialects.Scala211
+import scala.meta.tests.DiffAssertions
+
+class XmlSuite extends ParseSuite with DiffAssertions {
+  def tokenize(code: String): Tokens = {
+    val convert = scala.meta.inputs.Input.stringToInput
+    val tokenize = scala.meta.tokenizers.Tokenize.scalametaTokenize
+    val dialect = Scala211
+    code.tokenize(convert, tokenize, dialect).get
+  }
+
+  def tokensStructure(tokenized: Tokens): String = {
+    tokenized
+      .map(x => f"${x.structure}%22s ----> ${x.getClass}")
+      .mkString("\n")
+  }
+
+  test("simple xml literal - 1") {
+    assert(
+      tokensStructure(tokenize("<foo>bar</foo>")) ===
+        """|
+           |            BOF [0..0) ----> class scala.meta.tokens.Token$BOF
+           |                [0..0) ----> class scala.meta.tokens.Token$Xml$Start
+           |<foo>bar</foo> [0..14) ----> class scala.meta.tokens.Token$Xml$Part
+           |              [14..14) ----> class scala.meta.tokens.Token$Xml$End
+           |          EOF [14..14) ----> class scala.meta.tokens.Token$EOF
+           |""".stripMargin.trim.replace("QQQ", "\"\"\""))
+  }
+
+  test("simple xml literal - 2") {
+    assert(
+      tokensStructure(tokenize("<foo>bar</foo> ")) ===
+        """|
+           |            BOF [0..0) ----> class scala.meta.tokens.Token$BOF
+           |                [0..0) ----> class scala.meta.tokens.Token$Xml$Start
+           |<foo>bar</foo> [0..14) ----> class scala.meta.tokens.Token$Xml$Part
+           |              [14..14) ----> class scala.meta.tokens.Token$Xml$End
+           |              [14..15) ----> class scala.meta.tokens.Token$Space
+           |          EOF [15..15) ----> class scala.meta.tokens.Token$EOF
+           |""".stripMargin.trim.replace("QQQ", "\"\"\""))
+  }
+
+  // TODO(olafur) currently crashes, see https://github.com/scalameta/scalameta/pull/469/files/e2324548b3896d5e7b0993f4063316602d9583ff#r78293500
+  ignore("curlies inside scala expr") {
```

Comment:
@xeno-by I need to fix this failing test first. I don't know yet when I'll get to it.